dbux-data/src/applications/Application.js

```diff
@@ -125,7 +126,17 @@ export default class Application {
   getAppCommonAncestorPath() {
     const { staticProgramContexts } = this.dataProvider.collections;
     const paths = staticProgramContexts.getAllExisting().map(p => p.filePath);
-    return getCommonAncestorPath(...paths);
+    const commonAncestorPath = getCommonAncestorPath(...paths);
+    if (fs.lstatSync(commonAncestorPath).isDirectory()) {
```

Comment:
1. Let's try to avoid making any `fs` calls unless absolutely necessary.
2. `lstat` will most likely throw exceptions in all kinds of scenarios. Among others, it will most likely throw if the folder does not exist. (Not sure if that is a concern, but it is good to be aware of).

Can't we just check if `staticProgramContexts.getAllExisting().length` is 1 (or somesuch) instead?